src/app/modules/item/task/task-xd-pr.ts

```diff
@@ -0,0 +1,303 @@
+export function getTaskUrl(taskUrl : string, token : string, platform : string, prefix? : string, locale? : string) : string {
+  const channelId = (prefix || '') + getRandomID();
+  if (taskUrl.indexOf('?') == -1) {
+    // the idea is not to change the base url even if we change token, so we put token after #
+    taskUrl = taskUrl + '?';
+  } else {
+    taskUrl = taskUrl + '&';
+  }
+  let url = taskUrl
+      + 'sToken=' + encodeURIComponent(token)
+      + '&sPlatform=' + encodeURIComponent(platform)
+      + '&channelId=' + encodeURIComponent(channelId);
+  if (locale) {
+    url += '&sLocale=' + encodeURIComponent(locale);
+  }
+  return url;
+}
+
+/** Get Task object from an iframe */
+export function getTaskProxy(iframe : HTMLIFrameElement): Observable<Task> {
+  // Check every second whether the iframe loaded
+  return interval(1000)
+    .pipe(
+      // Abort after 300 seconds
+      map(
+        n => {
+          if (n > 300) {
+            throw Error("IFrame load waiting timeout");
+          }
+          return n;
+        }
+      ),
+
+      // Get contentWindow when it exists
+      filter(_ => !!iframe.contentWindow),
+      take(1),
+      map(_ => iframe.contentWindow as Window),
+
+      // Get the RxMessagingChannel from the contentWindow
+      switchMap((contentWindow: Window) =>
+        rxBuild({
+          window: contentWindow,
+          origin: "*",
+          scope: getUrlParameterByName('channelId', iframe.src),
+        })
+      ),
+
+      // Get the Task from the channel
+      map((chan: RxMessagingChannel) => new Task(chan))
+    );
+}
+
+
+/**
+ * Task object, created from an iframe DOM element
+ */
+export class Task {
+  platformSet = false;;
+  chan: RxMessagingChannel;
+
+  constructor(chan: RxMessagingChannel) {
+    this.chan = chan;
+  }
+
+  destroy(): void {
+    this.chan.destroy();
+  }
+
+  setPlatform(platform : Platform): void {
+    if (this.platformSet) {
+      throw new Error("Task already has a platform set");
+    }
+    this.chan.bind('platform.validate', (mode: string) => platform.validate(mode));
+    this.chan.bind('platform.getTaskParams', (keyDefault? : [string, TaskParamsValue]) => platform.getTaskParams(keyDefault));
+    this.chan.bind('platform.showView', (view : TaskView) => platform.showView(view));
+    this.chan.bind('platform.askHint', (hintToken : string) => platform.askHint(hintToken));
+    this.chan.bind('platform.updateDisplay', (data : TaskDisplayData) => platform.updateDisplay(data));
+    this.chan.bind('platform.openUrl', (url : string) => platform.openUrl(url));
+    this.chan.bind('platform.log', (data : TaskLog) => platform.log(data));
+
+    // Legacy calls
+    this.chan.bind('platform.updateHeight', (height : number) => platform.updateDisplay({ height: height }));
+  }
+
+  /**
+   * Task API functions
+   */
+  load(views : Object): Observable<void> {
+    return this.chan.call({
+      method: "task.load",
```

Comment:
Normally the linter should only allow single quotes, strange.

Reply:
Fixed.